modules/integration/tests-integration/tests/src/test/java/org/wso2/appserver/integration/tests/usermgt/UserManagementWithAdminUserTestCase.java

```diff
@@ -0,0 +1,408 @@
+		FlaggedName[] rolesOfUser = userManagementClient.getRolesOfUser(username, roleName, 10);
+		for (FlaggedName flaggedName : rolesOfUser) {
+			if (roleName.equals(flaggedName.getItemName())) {
+				assertTrue(flaggedName.getSelected(), "User don't have this role");
+				break;
+			}
+		}
+
+		userManagementClient.addRemoveRolesOfUser(username, null, new String[] { roleName });
+		rolesOfUser = userManagementClient.getRolesOfUser(username, roleName, 10);
+		for (FlaggedName flaggedName : rolesOfUser) {
+			if (roleName.equals(flaggedName.getItemName())) {
+				assertFalse(flaggedName.getSelected(), "Remove role failed");
+				return;
+			}
+		}
+		fail("Remove role failed");
+	}
+
+	@Test(groups = "wso2.as", description = "Add already existing user", dependsOnMethods = {
+			"testRemoveRoleFromUser" }, expectedExceptions = UserAdminUserAdminException.class,
+			expectedExceptionsMessageRegExp = "UserAdminUserAdminException")
+	public void testCreateExistingUser() throws Exception {
+		userManagementClient.addUser(username, password, null, username);
+	}
+
+	@Test(groups = "wso2.as", description = "Add already existing role", dependsOnMethods = {
+			"testCreateExistingUser" }, expectedExceptions = RemoteException.class,
+			expectedExceptionsMessageRegExp = "Role name:.*in the system. Please pick another role name.")
+	public void testCreateExistingRole() throws Exception {
+		userManagementClient.addRole(roleName, null, permissions);
+	}
+
+	@Test(groups = "wso2.as", description = "Get permissions of the Admin role")
+	public void testGetPermissionsOfRole() throws Exception {
+		UIPermissionNode rolePermissions = userManagementClient.getRolePermissions(TEST_NON_ADMIN_ROLE_NAME);
+		UIPermissionNode loginPermissionNode;
+		if (userMode == TestUserMode.SUPER_TENANT_ADMIN) {
+			loginPermissionNode = rolePermissions.getNodeList()[0].getNodeList()[1];
+		} else {
+			loginPermissionNode = rolePermissions.getNodeList()[1];
+		}
+		assertEquals(loginPermissionNode.getResourcePath(), "/permission/admin/login",
+		             "/permission/admin/login/ in not in the requested position");
+		assertTrue(loginPermissionNode.getSelected(),
+		           "/permission/admin/login/ is not available for " + TEST_NON_ADMIN_ROLE_NAME);
+	}
+
+	@Test(groups = "wso2.as", description = "Get list of users and check if that is equal to the no of users define in automation.xml")
+	public void testListAllUsers() throws Exception {
+		FlaggedName[] listOfUsers = userManagementClient.listAllUsers("*", 10);
+		// Get user count defined in automation.xml + last element is false when call through the UserAdminClient
+		if (userMode == TestUserMode.SUPER_TENANT_ADMIN) {
+			assertTrue(listOfUsers.length >= 6, "User count is differ from the expected count");
+		} else {
+			assertTrue(listOfUsers.length >= 5, "User count is differ from the expected count");
+		}
+	}
+
+	@Test(groups = "wso2.as", description = "Change password of current user defined in automation.xml")
```

Comment:
If a method in a test class as a depends on then as a practice need to define depends on for all other methods.